### PR TITLE
BR-143 Create CI Pipeline for codice-itest-api

### PR DESCRIPTION
**Pull request by TonyMorrison**
### What is this PR for?
Create a new Jenkinsfile pipeline that includes a Full Build stage, an Incremental Build stage for PRs and a Deploy stage

## Reviewers
@dcruver 
@shaundmorris

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'25'))
+        disableConcurrentBuilds()
+        timestamps()
+        skipDefaultCheckout()
+    }
+    triggers {
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
```
Comment (shaundmorris) [inline Jenkinsfile]: ❓ Is this comment correct? Looks like this runs weekly if it is master versus nightly.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        success {
+            slackSend color: 'good', message: "SUCCESS: ${JOB_NAME} ${BUILD_NUMBER}"
+        }
+        failure {
+            slackSend color: '#ea0017', message: "FAILURE: ${JOB_NAME} ${BUILD_NUMBER}. See the results here: ${BUILD_URL}"
+        }
+        unstable {
+            slackSend color: '#ffb600', message: "UNSTABLE: ${JOB_NAME} ${BUILD_NUMBER}. See the results here: ${BUILD_URL}"
+        }
+        cleanup {
```
Comment (shaundmorris) [inline Jenkinsfile:107]: ❔ Do we still need this cleanup section?

**Review by shaundmorris — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (TonyMorrison) [reply Jenkinsfile]: i can ask @dcruver about frequency

File: Jenkinsfile (hunk 2 above)
Comment (TonyMorrison) [reply Jenkinsfile:107]: perhaps not all, I removed the cleanup of the .m2 directory, the rest seems like good practice?

**Pull request merged**